go/consensus/tendermint/abci/mux.go

```diff
@@ -568,7 +544,23 @@ func (mux *abciMux) BeginBlock(req types.RequestBeginBlock) types.ResponseBeginB
 	}
 
 	response := mux.BaseApplication.BeginBlock(req)
-	response.Events = ctx.GetEvents()
+
+	// Check any events left from InitChain.
+	_, evJSON := mux.state.deliverTxTree.Get([]byte(stateKeyInitChainEvents))
+	if evJSON == nil {
+		// For the second and the following blocks, only return events from the
+		// application's BeginBlock calls.
+		response.Events = ctx.GetEvents()
+	} else {
+		// During the first block, also emit application events generated during
```

Comment:
Can we also enforce that this only happens in the first block and a non-nil value in `stateKeyInitChainEvents` is an error (panic) otherwise?